### PR TITLE
refactor(site): update agent status to include the lifecycle

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Updating the FE, specifically the <AgentStatus> component to handle the new agent's lifecycles.

Refs: #5749

### CODE REVIEW

**Review by mafredri — APPROVED** [review]
Very nice job @BrunoQuaresma! I tried it out locally and it looks like the behavior is on-point.

The only comment I have is that visually the "red" for error isn't very noticeable for me (not sure if my slight color blindness plays a role), but the warning color is much more noticeable/full, and the red looks faded (almost gray/pink).

Warning:
image
Error:
image

**Comment by mafredri** [comment]
@BrunoQuaresma @kylecarbs @bpmct should we label this as a breaking release?

Once this is deployed, and users update to provider v0.6.7+, users will start to see that workspaces time out during start (default 5 minutes). If they're using startup script to launch a service synchronously, then they'll run into it.

**Comment by kylecarbs** [comment]
Could you post a screenshot of the hover over the icon?

I think breaking is alright if we provide some good documentation to fix this, but not if we don't. FWIW we could tell users to use `screen` for long-running processes in the meantime, and it'll have roughly the same effect as before.

**Comment by mafredri** [comment]
@kylecarbs

image
image

We definitely want to improve documentation around this, we already mention checking logs @ the end of that [troubleshooting link](https://coder.com/docs/v2/latest/templates#troubleshooting-templates), but we can more explicitly mention the states and what log is relevant.

**Comment by mafredri** [comment]
Sure, let's improve the wording and perhaps add a second link 👍🏻. This is mostly a soft-break, though, as the experience is basically the same except for this visual indication here.

In the CLI this is going to be basically opt-in by upgrading provider and setting `delay_login_until_ready = true`.

**Comment by BrunoQuaresma** [comment]
@mafredri about the colors, it is a known issue. I'm going to try to fix that in a next PR.

File: site/src/pages/WorkspacePage/WorkspacePage.test.tsx [hunk 1]
```diff
@@ -284,65 +280,4 @@ describe("WorkspacePage", () => {
       })
     })
   })
-
-  describe("Resources", () => {
```
Comment (BrunoQuaresma) [inline site/src/pages/WorkspacePage/WorkspacePage.test.tsx:288]: I really think tests like these should live in Storybook. Integration tests should be used for user interactions IMO.

**Pull request merged**